### ISSUE TITLE
[Enhancement] [Probably for SM6] Instaneous Game Type change

### ISSUE THREAD

**Issue by JOELwindows7** [stepmania/stepmania#1049]
It means, changing game mode takes instaneous time without long time loading, the only delay is caused by if another game type was choosing different theme, otherwise it takes a second.

Good for __Auto set style__, not just having a gameplay style chosen in difficulty selection, but also you can switch gametype on the go in the difficulty selection too!
(__Note__: in case in the user's setting this gametype choses this theme and another choses other theme, if the player from this gametype entered the Select music and chose that other gametype, the theme won't be changed. If the user quitted the game, the gametype will still that this one had been chosen and the theme also unchanged.)

In conclusion, this system allows no delay gametype change.

**Comment by kyzentun** [comment]
There's a ton of groundwork that needs to be laid to make this possible.  It's been on the drawing board for a long time, and it's not going to happen for a long time.

**Comment by JOELwindows7** [comment]
@kyzentun I know it. this one is hard. requires entire code rewrite on some cpp something

**Comment by shakesoda** [comment]
Please don't tag your issues in the titles, we can take care of that.
